### ISSUE TITLE
Is synchronizing location to Redux store an anti-pattern?

### ISSUE THREAD

**Issue by mklimuk** [ReactTraining/react-router-addons-controlled#8]
Hi guys,

Thanks for the good job on the router & co!
I started playing around with react-router v4 and I got tempted by the idea of storing location in my redux store using the controlled router. But I soon realized that by putting location is store we introduce a second source of trough which is unnecessary. Location is a specific part of the application state that is coupled with the browser (at least in browser-based applications) and should stay right where it fits the most: on the browser side.

I was wondering what is your opinion about it and shouldn't we match storing location in store as an anti-pattern. Any thoughts?

**Comment by gaelduplessix** [comment]
As far as I understand, the idea of `ControlledRouter` is actually to have one source of truth: your redux state. Conceptually, the browser doesn't hold the history state anymore but only displays what the application gives to it, just like a controlled input. To change this state, you bind events just like you would use `onChange` to change the value of an input.
I think this is the overall vision but in practice this is a bit tricky because the browser does have internal state and doesn't provide all the need APIs to have a fully controlled history. For example, it can decide to redirect/change the URL without asking for permission to the app, so I'm not sure it's possible to have a 100% controlled history.

**Comment by Noitidart** [comment]
> As far as I understand, the idea of ControlledRouter is actually to have one source of truth: your redux state. Conceptually, the browser doesn't hold the history state anymore but only displays what the application gives to it, just like a controlled input. To change this state, you bind events just like you would use onChange to change the value of an input.

I agree with that and think it is inline with the Redux pattern. Location change is now treated as event. Specifically due to the `if (action === 'SYNC') {` in the `onChange` even of the router.
